File: go/wam/4994.go
const WamStatusCrosspostRequest = 4994 // channel: regular

type StatusCrosspostRequestEvent struct {
	CacSessionId string `wam:"cac_session_id"` // field 1
	CrosspostDestination CROSSPOSTDESTINATIONTYPE `wam:"crosspost_destination"` // field 2
	CrosspostErrorType string `wam:"crosspost_error_type"` // field 3
	CrosspostOrigin CROSSPOSTORIGINTYPE `wam:"crosspost_origin"` // field 4
	DefaultStatusPrivacySettings PRIVACYSETTINGSVALUETYPE `wam:"default_status_privacy_settings"` // field 5
	IsAutoCrosspostEnabledInSettings bool `wam:"is_auto_crosspost_enabled_in_settings"` // field 6
	IsAutoCrossposted bool `wam:"is_auto_crossposted"` // field 7
	StatusCrossPostPerPostStatusPrivacySetting PRIVACYSETTINGSVALUETYPE `wam:"status_cross_post_per_post_status_privacy_setting"` // field 8
	StatusCrosspostMediaType MEDIATYPE `wam:"status_crosspost_media_type"` // field 9
	StatusCrosspostResult CROSSPOSTRESULTTYPE `wam:"status_crosspost_result"` // field 10
	StatusId string `wam:"status_id"` // field 11
	StatusCrosspostEntryPoint string `wam:"status_crosspost_entry_point"` // field 12
	StatusCrosspostShareType STATUSCROSSPOSTSHARETYPEENUM `wam:"status_crosspost_share_type"` // field 13
	StatusCrosspostEventType string `wam:"status_crosspost_event_type"` // field 14
	StatusCrosspostFlowTraceId string `wam:"status_crosspost_flow_trace_id"` // field 15
	StatusCrosspostTraceId int64 `wam:"status_crosspost_trace_id"` // field 16
	UserJourneyEventMs int64 `wam:"user_journey_event_ms"` // field 17
}

File: rust/wam/4994.rs
pub const WAM_STATUS_CROSSPOST_REQUEST: u32 = 4994;

#[derive(Debug, Default)]
pub struct StatusCrosspostRequestEvent {
    /// field 1, wire `cac_session_id`
    pub cac_session_id: Option<String>,
    /// field 2, wire `crosspost_destination`
    pub crosspost_destination: Option<CROSSPOSTDESTINATIONTYPE>,
    /// field 3, wire `crosspost_error_type`
    pub crosspost_error_type: Option<String>,
    /// field 4, wire `crosspost_origin`
    pub crosspost_origin: Option<CROSSPOSTORIGINTYPE>,
    /// field 5, wire `default_status_privacy_settings`
    pub default_status_privacy_settings: Option<PRIVACYSETTINGSVALUETYPE>,
    /// field 6, wire `is_auto_crosspost_enabled_in_settings`
    pub is_auto_crosspost_enabled_in_settings: Option<bool>,
    /// field 7, wire `is_auto_crossposted`
    pub is_auto_crossposted: Option<bool>,
    /// field 8, wire `status_cross_post_per_post_status_privacy_setting`
    pub status_cross_post_per_post_status_privacy_setting: Option<PRIVACYSETTINGSVALUETYPE>,
    /// field 9, wire `status_crosspost_media_type`
    pub status_crosspost_media_type: Option<MEDIATYPE>,
    /// field 10, wire `status_crosspost_result`
    pub status_crosspost_result: Option<CROSSPOSTRESULTTYPE>,
    /// field 11, wire `status_id`
    pub status_id: Option<String>,
    /// field 12, wire `status_crosspost_entry_point`
    pub status_crosspost_entry_point: Option<String>,
    /// field 13, wire `status_crosspost_share_type`
    pub status_crosspost_share_type: Option<STATUSCROSSPOSTSHARETYPEENUM>,
    /// field 14, wire `status_crosspost_event_type`
    pub status_crosspost_event_type: Option<String>,
    /// field 15, wire `status_crosspost_flow_trace_id`
    pub status_crosspost_flow_trace_id: Option<String>,
    /// field 16, wire `status_crosspost_trace_id`
    pub status_crosspost_trace_id: Option<i64>,
    /// field 17, wire `user_journey_event_ms`
    pub user_journey_event_ms: Option<i64>,
}